src/useReactNativeZoomHandler.tsx

```diff
@@ -0,0 +1,173 @@
+// TODO:
+// There is a jump in the following situation:
+// Two finger pinch / drag -> 1 finger drag -> 2 finger drag
+// The focal offset is not being reapplied perfectly.
+
+import { useState } from 'react';
+import { useSharedValue, useAnimatedStyle } from 'react-native-reanimated';
+import { clamp } from './helper/clamp';
+import { pinchDragTranslate } from './helper/pinchDragTranslate';
+import { pinchTranslate } from './helper/pinchTranslate';
+
+export const useReactNativeZoomHandler = ({
```

Comment:
This looks good overall. One we might wanna consider is split this into 2 or more hooks as the logic is getting more complex with multiple gestures

Reply:
That's a good idea. I initially thought that these two gestures would need to need to be calculated together, but it seems like they are quite separate so that would work.